lib/ringsc.gi

```diff
@@ -403,14 +402,58 @@ InstallGlobalFunction( RingByStructureConstants, function( arg )
         NewType( Fam, IsSCRingObj and IsDenseCoeffVectorRep );
 
     SetCoefficientsFamily( Fam, ElementsFamily( FamilyObj( Integers ) ) );
-    # temporary
-    SetIsUFDFamily(Fam,false);
 
     # Make the generators and the ring.
     SetZero( Fam, ObjByExtRep( Fam, List( [ 1 .. n ], x -> 0 ) ) );
     gens:= Immutable( List( IdentityMat( n, Integers ),
                             x -> ObjByExtRep( Fam, x ) ) );
-    A:= RingByGenerators( gens );
+
+    # Hope to find a unit (and thus an identity)
+    one:=fail;
+    if oneinfo=true then
+      # find one, if possible
+      # Try generators first
+      i:=1;
+      while one=fail and i<=Length(gens) do
+        A:=[];
+        new:=gens[i];
+        repeat
+          Add(A,new);
+          new:=new*gens[i];
+        until new in A;
+        if new=gens[i] then
+          new:=A[Length(A)];
+          if ForAll(gens,x->x*new=x and new*x=x) then
+            one:=new;
+          fi;
+        fi;
+        i:=i+1;
+      od;
+
+      if one=fail then
+        # if not too big go through elements
+        if Product(moduli)<=10^5 then
```

Comment:
Meta question, out of curiosity, but: finding a unit element amounts to solving a linear system of equations over the integers (complicated a bit by the fact that we have these moduli in place). Why don't we just solve this?

(This is a honest question, not a veiled request/suggestion: I merely wonder if you tried this and it has problems; or if you decided to not do it to quickly get something working, even if incomplete. I am certainly not "demanding" that this be implemented in this PR; though it might be a good idea to add a comment suggesting that the general case could also be dealt with, if someone has need for it.)

Reply:
OK makes sense